presto-main/src/test/java/io/prestosql/execution/scheduler/TestSubnetTopology.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.execution.scheduler;
+
+import com.google.common.collect.ImmutableList;
+import org.testng.annotations.Test;
+
+import static io.prestosql.execution.scheduler.NetworkLocation.ROOT_LOCATION;
+import static io.prestosql.execution.scheduler.SubnetBasedTopology.AddressProtocol.IPV4;
+import static io.prestosql.execution.scheduler.SubnetBasedTopology.AddressProtocol.IPV6;
+import static io.prestosql.spi.HostAddress.fromString;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertThrows;
+
+public class TestSubnetTopology
+{
+    @Test
+    public void testSubnetTopologyIpv4()
+    {
+        SubnetBasedTopology topology = new SubnetBasedTopology(ImmutableList.of(24, 25, 27), IPV4);
+        assertEquals(
+                topology.locate(fromString("192.168.0.172")),
```

Comment:
Don't static import `fromString` since it's not clear from the name alone what it is producing